src/test/scala/TextToStructuredDataSpec.scala

```diff
@@ -0,0 +1,140 @@
+    )), actual.schema)
+
+    assertSmallDataFrameEquality(actual, expected)
+  }
+
+  it("Language check") {
+    val source = spark.createDataset[String](List(
+      "List(RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, en, true)",
+      "List(RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, de, true)",
+      "List(RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, jp, true)",
+      "List(RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, fr, true)",
+      "List(RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, ru, false)"
+    ))
+
+    val actual = getStructuredDataFromText(source)
+
+    val expected = spark.createDataFrame(spark.sparkContext.makeRDD(List(
+      Row("RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","en",1,0.toLong),
+      Row("RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","de",1,1.toLong),
+      Row("RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","jp",1,2.toLong),
+      Row("RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","fr",1,3.toLong),
+      Row("RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","ru",0,4.toLong)
+    )), actual.schema)
+
+    assertSmallDataFrameEquality(actual, expected)
+  }
+
+  it("New lines check") {
+    val source = spark.createDataset[String](List(
+      """List(
+        |RT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4, en, true)""".stripMargin,
+      """List($btc #bitcoin #cryptolife
+        |
+        |#cryptofamily Good morning, Bitcoin has good support now,
+        |he might not instantly jump up,…
+        |https://t.co/ok8vMxuE17, en, false)""".stripMargin,
+      """List(Earn bitcoin on a daily basis!
+        |1. Follow @slidecoin
+        |2. Complete instructions in pinned tweet, en, false)""".stripMargin
+    ))
+
+    val actual = getStructuredDataFromText(source)
+
+    val expected = spark.createDataFrame(spark.sparkContext.makeRDD(List(
+      Row("\nRT @emy_wng: Thanks Bloomberg for featuring @Ripple and #MoneyTap! https://t.co/WDuJnD1tq4","en",1,0.toLong),
+      Row("$btc #bitcoin #cryptolife\n\n#cryptofamily Good morning, Bitcoin has good support now,\nhe might not instantly jump up,…\nhttps://t.co/ok8vMxuE17","en",0,1.toLong),
+      Row("Earn bitcoin on a daily basis!\n1. Follow @slidecoin\n2. Complete instructions in pinned tweet","en",0,2.toLong)
+    )), actual.schema)
+
+    actual.foreach(row => {println(row.toString())})
+    expected.foreach(row => {println(row.toString())})
```

Comment:
please don't do printing to sys out :) 
use logging with debug level if required. 
and I doubt that this particular case requires it at all.

Reply:
Just forgot to delete my debug output.